portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -1,161 +1,80 @@
- * @returns {Promise<object>} - azure storage details with folder & file name, full path & url.
- */
-const saveFileToAzure = async (file, bankId) => {
-  try {
-    console.info(`Attempting to save utilisation report to Azure for bank: ${bankId}`);
-    const { originalname, buffer } = file;
-
-    const fileInfo = await uploadFile({
-      fileshare: FILESHARES.UTILISATION_REPORTS,
-      folder: bankId,
-      filename: formatFilenameForSharepoint(originalname),
-      buffer,
-      allowOverwrite: true,
-    });
-
-    if (!fileInfo || fileInfo.error) {
-      throw new Error(`Failed to save utilisation report to Azure - ${fileInfo?.error?.message ?? 'cause unknown'}`);
-    }
-
-    console.info(`Successfully saved utilisation report to Azure for bank: ${bankId}`);
-    return fileInfo;
-  } catch (error) {
-    console.error('Failed to save utilisation report to Azure ', error);
-    throw error;
-  }
-};
+  return reportsForPeriod[0];
+}
 
 const uploadReportAndSendNotification = async (req, res) => {
   try {
     const { file } = req;
 
-    const { formattedReportPeriod, reportData, reportPeriod, user } = req.body;
-    const parsedReportData = JSON.parse(reportData);
-    const parsedUser = JSON.parse(user);
-    const parsedReportPeriod = JSON.parse(reportPeriod);
-
     if (!file) {
       return res.status(400).send();
     }
 
+    const { formattedReportPeriod, reportData, reportPeriod, user } = req.body;
+    const parsedReportData = JSON.parse(reportData);
+    const parsedUser = JSON.parse(user);
+    const parsedReportPeriod = JSON.parse(reportPeriod);
     const bankId = parsedUser?.bank?.id;
 
-    const existingReports = await api.getUtilisationReports(bankId, {
-      reportPeriod: parsedReportPeriod,
-    });
-
-    if (existingReports.length !== 1) {
-      return res.status(500).send(`Expected 1 report but found ${existingReports.length} with bank ID ${bankId} and report period '${parsedReportPeriod}'`);
-    }
-
-    const existingReport = existingReports[0];
-
-    if (existingReport.status !== UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED) {
-      return res
-        .status(500)
-        .send(
-          `Expected report to be in '${UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED}' state (was actually in '${existingReport.status}' state)`,
-        );
+    const report = await getReportForPeriod(bankId, parsedReportPeriod);
+    validateReportIsInReportNotReceivedState(report);
+    const azureFileInfo = await saveUtilisationReportFileToAzure(file, bankId);
+    const { dateUploaded } = await api.saveUtilisationReport(report.reportId, parsedReportData, parsedUser, azureFileInfo);
+
+    try {
+      await sendUtilisationReportUploadNotificationEmailToUkefGefReportingTeam(
+        parsedUser?.bank?.name,
+        formattedReportPeriod,
+      );
+    } catch (error) {
+      console.error('Failed to send report upload notification to ukef gef reporting team %o', error);
```

Comment:
Do you think it is worthwhile for us to add a comment above this try-catch to explain why we don't want everything to fail if this throws?

Reply:
I guess I didn't add one because to me it feels fairly clear what the intention is and I am quite used to this being the pattern around downstream secondary actions. It is covered by a test case that you would have to intentionally remove to change this. That said, if this looks confusing to you as a reader then I will add a comment as that would be the only reason for a comment and as the author it is perhaps not down to me to solely judge that.

Reply:
It doesn't look confusing to me - I just wanted to know whether you thought it would be worth including a comment or not. More than happy to not include one as I do think it is quite clear as is :)